R/MultilabelNestedStackingWrapper.R

```diff
@@ -0,0 +1,107 @@
+#' @title Use nested stacking method to create a multilabel learner.
+#'
+#' @description
+#' Every learner which is implemented in mlr and which supports binary
+#' classification can be converted to a wrapped nested stacking multilabel learner.
+#' Nested stacking trains a binary classifier for each label following a given order. In training phase,
+#' the feature space of each classifier is extended with predicted label information (by cross validation) 
+#' of all previous labels in the chain. 
+#' During the prediction phase, predicted labels are obtained by the classifiers, which have been learned on 
+#' all training data.
+#'
+#' Models can easily be accessed via \code{\link{getLearnerModel}}.
+#'
+#' @param order The chain order. E.g. for \code{m} labels, this must be a permutation of \code{1:m}. Default is \code{"random"}, i.e. \code{sample(1:m)}.
+#' @param  cv.folds The number of folds for the inner cross validation method to predict labels for the augmented feature space. Default is \code{2}.
+#' @template arg_learner
+#' @template ret_learner
+#' @references
+#' Montanes, E. et al. (2013),
+#' \emph{Dependent binary relevance models for multi-label classification}
+#' Artificial Intelligence Center, University of Oviedo at Gijon, Spain.
+#' @family wrapper
+#' @family multilabel
+#' @export
+#' @examples
+#' d = getTaskData(yeast.task)
+#' # drop some labels so example runs faster
+#' d = d[, c(1:3, 15:117)]
+#' task = makeMultilabelTask(data = d, target = c("label1", "label2", "label3"))
+#' lrn = makeMultilabelNestedStackingWrapper("classif.rpart", order = 1:3)
+#' lrn = setPredictType(lrn, "prob")
+#' # train, predict and evaluate
+#' mod = train(lrn, task)
+#' pred = predict(mod, task)
+#' p = performance(pred)
+#' performance(pred, measure = hamloss)
+#' getMultilabelBinaryPerformances(pred, measures = list(mmce, auc))
+#' # above works also with predictions from resample!
+makeMultilabelNestedStackingWrapper = function(learner, order = "random", cv.folds = 2) {
+  learner = checkLearner(learner, type = "classif")
+  if ("twoclass" %nin% learner$properties) {
```

Comment:
Generally, there could be classification learners without property "twoclass".
One example are "oneclass" classification learners (we currently don't have any of those integrated, but we have a learner property "oneclass").

For the check I would use `hasLearnerProperties` (see e.g. `mlr:::checkLearnerBeforeTrain`)

@berndbischl : Philipp is right that this check is not in other wrappers that require a learner with property "twoclass", for example makeOversampleWrapper or makeMulticlassWrapper. What is the reason?